pinot-common/src/main/java/com/linkedin/pinot/common/utils/EqualityUtils.java

```diff
@@ -17,125 +17,112 @@
 
 import java.util.ArrayList;
 import java.util.Arrays;
-import java.util.Collection;
 import java.util.Collections;
 import java.util.List;
-import java.util.Map;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
 
 
 /**
  * Various utilities to be used in implementing equals and hashCode.
  *
+ * Keep isEqual() for primitive type for historical reason, but try not using them.
  */
 public class EqualityUtils {
   private EqualityUtils() {
   }
 
+  @Deprecated
   public static boolean isEqual(int left, int right) {
     return left == right;
   }
 
+  @Deprecated
   public static boolean isEqual(long left, long right) {
     return left == right;
   }
 
+  @Deprecated
   public static boolean isEqual(float left, float right) {
-    return Float.compare(left, right) == 0;
+    return left == right;
```

Comment:
The contract for `equals()` is different than floating point equality. While NaN != NaN for floating point math, it should be considered equivalent for the purposes of object equality. 

From Effective Java Chapter 3 Item 7:
> For primitive fields whose type is not float or double, use the == operator for comparisons; for object reference fields, invoke the equals method recursively; for float fields, translate to int values using Float.floatToIntBits and compare the int values using the == operator; for double fields, translate to long values using Double.doubleToLongBits
and compare the long values using the == operator. (The special treatment of float and double fields is made necessary by the existence of Float.NaN, -0.0f, and the analogous double constants; see the Float.equals documentation for details.) For array fields, apply these guidelines to each element.